src/components/AnalysisPage/3dModelViewer/3dModelViewer.jsx

```diff
@@ -15,67 +15,48 @@ const ModelViewer = ({
   setShowToggle,
 }) => {
   const canvasRef = useRef(null);
-  const [objectURL, setObjectURL] = useState(null);
-  const [fileType, setFileType] = useState(null);
+  const sceneRef = useRef(null);
```

Comment:
I'm also not an expert in react Refs.

But here its a reference to the setUpThreeScene function call.

To the best of my knowledge

Using ref lets us load and clear new models into the scene without having to re-setup the scene every time a new model loads or clears. Where if we were using a modelViewerSceneHandler state, it would trigger a re-render on each change and we would be calling setUpThreeScene over and over again (like we were before i believe, causing that black flash).

.current is where the reference is located in sceneRef. Here is the output of `console.log(sceneRef)` 

![image](https://github.com/rknizzle/propermesh-view/assets/111093132/51c2ea6c-e6f5-4b70-86e2-5fcdaa50ca7f)